### PR TITLE
Better Farmbot

### PR DESCRIPTION
**Pull request by Kurfursten**
Properly updates when you press a button or swipe ID
Properly accepts water and fertilizer
Properly waits instead of spamming actions (e.g.: refill at sink)
Refills water in a reasonable amount of time
Will use just 10u of fertilizer at a time instead of dumping a full bottle

🆑 
* bugfix: After years, farmbots are fixed! Now they can do EVERYTHING except actually move to their target! Astar fix when?

### CODE REVIEW

File: code/game/machinery/bots/farmbot.dm [hunk 1]
```diff
@@ -187,7 +187,7 @@
 	src.icon_state = "[src.icon_initial][src.on]"
 	target = null
 	mode = FARMBOT_MODE_WAITING //Give the emagger a chance to get away! 15 seconds should be good.
-	spawn(150)
+	spawn(15 SECONDS)
 		mode = 0
```
Comment (Kurfursten) [inline code/game/machinery/bots/farmbot.dm]: no longer necessary because of the frustration rework, also several srcs to remove above this

Comment (Eneocho) [reply code/game/machinery/bots/farmbot.dm]: can it even attack or do anything evil when emagged if it can't move to its target

Comment (Kurfursten) [reply code/game/machinery/bots/farmbot.dm]: Yeah it avoids moving at all and just throws fertilizer bottles

Comment (Eneocho) [reply code/game/machinery/bots/farmbot.dm]: those bottles should be ever so slightly explosive
ammonium nitrate and what not

**Comment by SonixApache** [comment]
Do they still go berserk randomly?

**Comment by Kurfursten** [comment]
I don't THINK I touched anything related to automatic berserk. Sounds like it could be related to brand intelligence...? I recall what you're saying from many years ago but it has been so long since I saw it happen that I can't be sure

**Pull request merged**